src/workloads/sharding/ReshardCollectionReadHeavy.yml

```diff
@@ -0,0 +1,186 @@
+
+  # Note that the exact document size may exceed ApproxDocumentSize because of field names and other
+  # fields in the document.
+  ApproxDocumentSize: &ApproxDocumentSize 2000  # = 2kB
+  ApproxDocumentSize50Pct: &ApproxDocumentSize50Pct 1000  # = 1kB
+  DocumentCount: &DocumentCount 8_000_000  # for an approximate total of 12GB
+  NumChunks: &NumChunks 240                # with approximately 50MB per chunk
+
+  ShardKeyValueMin: &ShardKeyValueMin 1
+  # We use 100x the number of chunks so that the $sample aggregation to choose the new split points
+  # has a wide range of distinct values to choose from. It could be even larger.
+  ShardKeyValueMax: &ShardKeyValueMax 24_000
+
+  ReadOperations: &ReadOperations
+  - OperationName: findOne
+    OperationCommand:
+      Filter: {_id: {^RandomInt: {min: 1, max: *DocumentCount}}}
+
+  WriteOperations: &WriteOperations
+  - OperationName: updateOne
+    OperationCommand:
+      Filter: {_id: {^RandomInt: {min: 1, max: *DocumentCount}}}
+      Update: {$inc: {counter: 1}}
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 201
+  # The reshardCollection command is expected to take a long to complete so we give the actor
+  # running it a much higher socket timeout.
+  ReshardCollection:
+    QueryOptions:
+      maxPoolSize: 1
+      socketTimeoutMS: 36_000_000  # = 10 hours
+
+Actors:
+- Name: CreateShardedCollection
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: admin
+    Operations:
+    - OperationMetricsName: EnableSharding
+      OperationName: AdminCommand
+      OperationCommand:
+        enableSharding: *Database
+    - OperationMetricsName: ShardCollection
+      OperationName: AdminCommand
+      OperationCommand:
+        shardCollection: *Namespace
+        # Hashed sharding will pre-split the chunk ranges and evenly distribute them across all of
+        # the shards.
+        key: {oldKey: hashed}
+        numInitialChunks: *NumChunks
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: LoadInitialData
+  Type: MonotonicSingleLoader
+  Threads: 100
+  Phases:
+  - *Nop
+  - Repeat: 1
+    BatchSize: 1000
+    DocumentCount: *DocumentCount
+    Database: *Database
+    Document:
+      oldKey: {^RandomInt: {min: *ShardKeyValueMin, max: *ShardKeyValueMax}}
+      newKey: {^RandomInt: {min: *ShardKeyValueMin, max: *ShardKeyValueMax}}
+      counter: 0
+      padding: {^FastRandomString: {length: {^RandomInt: {min: *ApproxDocumentSize50Pct, max: *ApproxDocumentSize}}}}
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: ReshardCollection
+  Type: AdminCommand
+  Threads: 1
+  ClientName: ReshardCollection
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    Database: admin
+    Operations:
+    - OperationMetricsName: ReshardCollection
+      OperationName: AdminCommand
+      OperationCommand:
+        reshardCollection: *Namespace
+        key: {newKey: 1}
+  - *Nop
+
+- Name: ReadCollectionBeingResharded
+  Type: CrudActor
+  Threads: 95
```

Comment:
I don't think this will give you 95% read and 5% write. Reader threads progress faster than writer threads, and that'd result in having more than 95% of total operations being reads. I think we'd want every thread to perform a write after every 19 read operations to emulate the desired mixture of CRUD operations.

Reply:
Hmm, I thought over Slack you had suggested splitting up the reader clients from the writer clients? Should I change ReshardCollectionMixed.yml because it is using 50 threads for findOne and 50 threads for updateOne.

Reply:
I think it's okay to split the threads as long as we use the rate-limiter to enforce the ratio. Otherwise, I'm not sure if this will give us the ratio we want. How do you feel about using the rate-limiter here?

Reply:
It isn't possible to specify `GlobalRate` with `Blocking: None`. The DocumentThroughput values in my patch build suggest that we're closer to 2-4% write. I don't feel attached to the particular ratio other than wanting "almost entirely reads with some writes".

https://spruce.mongodb.com/task/sys_perf_linux_shard_lite_all_feature_flags_reshard_collection_read_heavy_patch_1d8a437c4d4f6ebf4c51546514bf8c4da99046fb_60cbf2fa2fbabe7e4152d83e_21_06_18_01_13_36/trend-charts?execution=2